tests/functional/v2/server/testListMetrics.js

```diff
@@ -0,0 +1,150 @@
+        'post',
+        `http://localhost:8100/${level}`,
+        body,
+        { json: true },
+    );
+}
+
+async function ingestEvents(events) {
+    return events.length === await warp10.ingest('utapi.event', events);
+}
+
+function opsToResp(operations) {
+    return Object.entries(operations)
+        .reduce((prev, [key, value]) => {
+            prev[operationToResponse[key]] = value;
+            return prev;
+        }, { ...emptyOperationsResponse });
+}
+
+const testCases = [
+    {
+        desc: 'for a single resource',
+        args: { [uuid.v4()]: { [uuid.v4()]: [uuid.v4()] } },
+    },
+    {
+        desc: 'for multiple resources',
+        args: {
+            [uuid.v4()]: {
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+            },
+            [uuid.v4()]: {
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+            },
+            [uuid.v4()]: {
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+                [uuid.v4()]: [uuid.v4(), uuid.v4(), uuid.v4()],
+            },
+        },
+    },
+];
+
+describe('Test listMetric', function () {
+    this.timeout(10000);
+    testCases.forEach(testCase => {
+        describe(testCase.desc, () => {
+            let totals;
+            before(async () => {
+                const { events, totals: _totals } = generateCustomEvents(
+                    getTs(-360),
+                    getTs(-60),
+                    1000,
+                    testCase.args,
+                );
+                totals = _totals;
+                assert(await ingestEvents(events));
+            });
+
+
+            const accounts = [];
+            const users = [];
+            const buckets = [];
+
+            Object.entries(testCase.args)
+                .forEach(([account, _users]) => {
+                    accounts.push(account);
+                    Object.entries(_users).forEach(([user, _buckets]) => {
+                        users.push(user);
+                        buckets.push(..._buckets);
+                    });
+                });
+
+            const metricQueries = {
+                accounts,
+                users,
+                buckets,
+            };
+
+            Object.entries(metricQueries)
+                .forEach(query => {
+                    const [level, resources] = query;
+                    it(`should get metrics for ${level}`, async () => {
+                        const resp = await listMetrics(...query, getTs(-500), getTs(0));
+
+                        if (resources.length === 1) {
+                            // If only one resource is requested the response is an object
```

Comment:
cf. my earlier comment about API consistency (always returning an array might be better IMHO unless the caller is able to pass an object for a single request for example, in which case we can return a single object since the contract is unambiguous).

As an example of where it can be problematic: if the API caller takes its request array programmatically, it might be annoyed to have an object response instead of an array in case the computed array only has a single element in it.